### ISSUE TITLE
The columns endpoint results in a 500, possibly metadata related

### ISSUE THREAD

**Issue by pavish** [centerofci/mathesar#1776]
## Description

Endpoint: `http://localhost:8000/api/db/v0/tables/<table_id>/columns/`

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/api/db/v0/tables/5/columns/?limit=500

Django Version: 3.1.14
Python Version: 3.9.8
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'rest_framework',
 'django_filters',
 'django_property_filter',
 'mathesar']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback (most recent call last):
  File "/code/mathesar/models/base.py", line 675, in __getattribute__
    return super().__getattribute__(name)

During handling of the above exception ('Column' object has no attribute 'primary_key'), another exception occurred:
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/sql/base.py", line 1167, in __getattr__
    return self._index[key]

The above exception ('nspname') was the direct cause of the following exception:
  File "/code/mathesar/models/base.py", line 675, in __getattribute__
    return super().__getattribute__(name)
  File "/code/mathesar/models/base.py", line 696, in _sa_column
    return self.table.sa_columns[self.name]
  File "/code/mathesar/models/base.py", line 362, in sa_columns
    return self._enriched_column_sa_table.columns
  File "/code/mathesar/models/base.py", line 350, in _enriched_column_sa_table
    table=self._sa_table,
  File "/code/mathesar/state/cached_property.py", line 62, in __get__
    new_value = self.original_get_fn(instance)
  File "/code/mathesar/models/base.py", line 331, in _sa_table
    sa_table = reflect_table_from_oid(
  File "/code/db/tables/operations/select.py", line 23, in reflect_table_from_oid
    tables = reflect_tables_from_oids([oid], engine, metadata=metadata, connection_to_use=connection_to_use)
  File "/code/db/tables/operations/select.py", line 29, in reflect_tables_from_oids
    get_map_of_table_oid_to_schema_name_and_table_name(
  File "/code/db/tables/operations/select.py", line 59, in get_map_of_table_oid_to_schema_name_and_table_name
    select(pg_namespace.c.nspname, pg_class.c.relname, pg_class.c.oid)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/sql/base.py", line 1169, in __getattr__
    util.raise_(AttributeError(key), replace_context=err)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception

During handling of the above exception (nspname), another exception occurred:
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/sql/base.py", line 1167, in __getattr__
    return self._index[key]

The above exception ('nspname') was the direct cause of the following exception:
  File "/code/mathesar/models/base.py", line 675, in __getattribute__
    return super().__getattribute__(name)
  File "/code/mathesar/models/base.py", line 696, in _sa_column
    return self.table.sa_columns[self.name]
  File "/code/mathesar/models/base.py", line 362, in sa_columns
    return self._enriched_column_sa_table.columns
  File "/code/mathesar/models/base.py", line 350, in _enriched_column_sa_table
    table=self._sa_table,
  File "/code/mathesar/state/cached_property.py", line 62, in __get__
    new_value = self.original_get_fn(instance)
  File "/code/mathesar/models/base.py", line 331, in _sa_table
    sa_table = reflect_table_from_oid(
  File "/code/db/tables/operations/select.py", line 23, in reflect_table_from_oid
    tables = reflect_tables_from_oids([oid], engine, metadata=metadata, connection_to_use=connection_to_use)
  File "/code/db/tables/operations/select.py", line 29, in reflect_tables_from_oids
    get_map_of_table_oid_to_schema_name_and_table_name(
  File "/code/db/tables/operations/select.py", line 59, in get_map_of_table_oid_to_schema_name_and_table_name
    select(pg_namespace.c.nspname, pg_class.c.relname, pg_class.c.oid)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/sql/base.py", line 1169, in __getattr__
    util.raise_(AttributeError(key), replace_context=err)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception

During handling of the above exception (nspname), another exception occurred:
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 826, in __getattr__
    return getattr(self.comparator, key)

The above exception ('Comparator' object has no attribute '_sa_column') was the direct cause of the following exception:
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 466, in handle_exception
    response = exception_handler(exc, context)
  File "/code/mathesar/exception_handlers.py", line 55, in mathesar_exception_handler
    raise exc
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/mixins.py", line 38, in list
    queryset = self.filter_queryset(self.get_queryset())
  File "/code/mathesar/api/db/viewsets/columns.py", line 33, in get_queryset
    queryset = Column.objects.filter(table=self.kwargs['table_pk']).order_by('attnum')
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/code/mathesar/models/base.py", line 70, in get_queryset
    make_sure_initial_reflection_happened()
  File "/code/mathesar/state/base.py", line 8, in make_sure_initial_reflection_happened
    reset_reflection()
  File "/code/mathesar/state/base.py", line 27, in reset_reflection
    _trigger_django_model_reflection()
  File "/code/mathesar/state/base.py", line 31, in _trigger_django_model_reflection
    reflect_db_objects(metadata=get_cached_metadata())
  File "/code/mathesar/state/django.py", line 44, in reflect_db_objects
    reflect_columns_from_tables(tables, metadata=metadata)
  File "/code/mathesar/state/django.py", line 123, in reflect_columns_from_tables
    models._compute_preview_template(table)
  File "/code/mathesar/models/base.py", line 859, in _compute_preview_template
    if column.primary_key:
  File "/code/mathesar/models/base.py", line 681, in __getattribute__
    return getattr(self._sa_column, name)
  File "/code/mathesar/models/base.py", line 681, in __getattribute__
    return getattr(self._sa_column, name)
  File "/code/mathesar/models/base.py", line 681, in __getattribute__
    return getattr(self._sa_column, name)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 828, in __getattr__
    util.raise_(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception

Exception Type: AttributeError at /api/db/v0/tables/5/columns/
Exception Value: Neither 'MathesarColumn' object nor 'Comparator' object has an attribute '_sa_column'
```

**Comment by pavish** [comment]
I'm closing this issue since I'm not able to reproduce this with a fresh setup.

**Comment by mathemancer** [comment]
For context, I believe this was fixed by #1775  .
